api/server/router/container/container.go

```diff
@@ -60,6 +60,7 @@ func (r *containerRouter) initRoutes() {
 		// PUT
 		local.NewPutRoute("/containers/{name:.*}/archive", r.putContainersArchive),
 		// DELETE
+		local.NewDeleteRoute("/containers/_all", r.deleteAllContainers),
```

Comment:
I think it would be more intuitive to not use the `_all` endpoint and have `/containers/` and `/containers` be the same. 

My main concern is that existing clients, including the Docker cli, probably rely on the error that's being returned when no container name is passed. Maybe the `_all` endpoint or `?all=1` flag, plus some kind of deprecation warning and documentation update, would be a good temporary solution?